CSLabs.Api/Controllers/HealthCheckController.cs

```diff
@@ -26,14 +28,53 @@ public async Task<IActionResult> TestProxmoxConnection()
             var hypervisors = await DatabaseContext.Hypervisors
                 .Include(h => h.HypervisorNodes)
                 .ToListAsync();
-            foreach (var hypervisor in hypervisors)
+            var systemStatus = DatabaseContext.SystemStatuses.First(); // Get current system status
+            try
             {
-                var api = ProxmoxManager.GetProxmoxApi(hypervisor.HypervisorNodes.First());
-                foreach (var node in hypervisor.HypervisorNodes)
+                foreach (var hypervisor in hypervisors)
                 {
-                    await api.GetNodeStatus(node);
+                    var nodesUp = 0;
+                    var api = ProxmoxManager.GetProxmoxApi(hypervisor.HypervisorNodes.First());
+
+                    try 
+                    {
+                        var clusterStatus = api.GetClusterStatus();
+                        if (!clusterStatus.Result.Quorate)
+                        {
+                            throw new NoQuorumException();
+                        }
+                    }
+                    catch (ProxmoxRequestException e)
+                    {
+                        // GetClusterStatus failed, so no nodes are up 
+                        systemStatus.HypervisorNodesUp = 0;
+                        await DatabaseContext.SaveChangesAsync();
+                    }
+                    
+                    // check the status of all the nodes and count how many are up
+                    foreach (var node in hypervisor.HypervisorNodes)
+                    {
+                        try
+                        {
+                            await api.GetNodeStatus(node);
+                        }
+                        catch (ProxmoxRequestException e)
+                        {
+                    
+                        }
+
+                        nodesUp++;
```

Comment:
Move notes up into the try otherwise an exception here won't  reduce the number of nodes up.